graphql-dgs/build.gradle.kts

```diff
@@ -23,6 +23,7 @@ dependencies {
 
     implementation("org.springframework:spring-web")
     implementation("org.springframework:spring-context")
+    implementation("org.springframework.boot:spring-boot")
```

Comment:
Think @paulbakker was interested on keeping it SB agnostic.

Reply:
So that, coupled w/ the blast radius of commit#2 - I am going to move these out of `dgs-graphql` and into each appropriate module (the fine-grained approach). I will put that work in a separate PR so we can compare/contrast them side/side.

Reply:
We have some (rare, but real) use cases where we have to bootstrap things manually, without using Spring Boot. So ideally, the core module should not depend on spring-boot. 

However, it's not a complete show stopper; if it's really problematic we can consider it.